engine/crates/composition/src/compose/entity_interface.rs

```diff
@@ -0,0 +1,181 @@
+use super::*;
+use crate::composition_ir as ir;
+use std::collections::{BTreeMap, BTreeSet};
+
+pub(crate) fn merge_entity_interface_definitions(
```

Comment:
Maybe some parts of this could be broken up in other functions. It wouldn't necessarily be any simpler, though.